template.js

```diff
@@ -69,35 +69,56 @@ if (isLoggingEnabled) {
   );
 }
 
-sendHttpRequest(
-  postUrl,
-  (statusCode, headers, body) => {
-    if (isLoggingEnabled) {
-      logToConsole(
-        JSON.stringify({
-          Name: 'LinkedIn',
-          Type: 'Response',
-          TraceId: traceId,
-          EventName: postBody.eventId,
-          ResponseStatusCode: statusCode,
-          ResponseHeaders: headers,
-          ResponseBody: body
-        })
-      );
-    }
+// perform validation check on presence of 1/4 of the required IDs. If at least 1 ID is present, make the API call. If no IDs are present, log the warning and no call is made
+if (validateUserData()){
+  sendConversionToLinkedIn();
+} else {
+  logToConsole('No conversion event was sent to CAPI. You must set 1 out of the 4 acceptable IDs (SHA256_EMAIL, LINKEDIN_FIRST_PARTY_ADS_TRACKING_UUID, ACXIOM_ID, ORACLE_MOAT_ID) to resolve this issue or make certain to send both firstName and lastName.');
 
-    if (statusCode >= 200 && statusCode < 300) {
-      data.gtmOnSuccess();
-    } else {
-      data.gtmOnFailure();
-    }
-  },
-  {
-    headers: postHeaders,
-    method: 'POST'
-  },
-  JSON.stringify(postBody)
-);
+  data.gtmOnFailure();
+}
+
+function validateUserData() {
+  if (postBody.user.userIds.length > 0) {
+    return true;
+  }
+
+  return postBody.user.userInfo.firstName != "" &&
```

Comment:
Maybe better to change to 

postBody.user.userInfo.firstName && postBody.user.userInfo.lastName

As they can be null or undefined

Reply:
return postBody.user.userInfo.firstName && postBody.user.userInfo.lastName;

Reply:
According to the implementation, it should always be a string, but I get your point 🙂 
```javascript
function getUserFirstName() {
  return (
    userInfoOverride.firstName ||
    eventData.firstName ||
    eventData.FirstName ||
    eventData.nameFirst ||
    eventData.first_name ||
    user_data.first_name ||
    user_address.first_name ||
    ''
  );
}

function getUserLastName() {
  return (
    userInfoOverride.lastName ||
    eventData.lastName ||
    eventData.LastName ||
    eventData.nameLast ||
    eventData.last_name ||
    user_data.last_name ||
    user_address.last_name ||
    ''
  );
}
```